docs/builtins.md

```diff
@@ -0,0 +1,225 @@
+Input
+```evy:input
+Mary Jackson
+```
+
+Output
+```evy:output
+Hello, Mary Jackson!
+```
+
+#### Reference 
+
+    read:string
+
+The `read` function returns a string that contains the line of input
+that the user entered up until, excluding the newline character. It is
+a blocking functions, which means that it will not return until the
+user has entered a line of input and pressed the Enter key.
+
+In a browser environment `read` typically reads from a DOM Textarea
+Element wired to the evy runtime. When running evy from the command
+line interface, `read` reads from standard in.
+
+### `cls`
+
+`cls` clears the output area of all printed text.
+
+#### Example
+```evy
+print "Hello"
+sleep 1
+cls
+print "Bye"
+```
+
+Output
+```evy:output
+Bye
+```
+
+#### Reference 
+
+    cls
+
+The `cls` function clears the output area. In a browser environment
+`cls` typically clears the DOM Element wired to the evy runtime which
+is used for printing. When running evy from the command line interface,
+`cls` clears the terminal, similar to the Unix `clear` or Windows `cls`
+commands.
+
+### `printf`
+
+`printf` stands for print formatted.
+
+`printf` prints its arguments to the output area according to a _format_
+string. The format string is the first argument, and it
+contains _specifiers_. Specifiers start with a percent sign `%`. They
+tell the `printf` function how and where to print the remaining
+arguments inside the format string. The rest of the format string is
+printed to the output area without changes. 
+
+Here are some valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| `%v`      | the argument in its default format  |
+| `%q`      | a double-quoted string |
+| `%%`      | a percent sign `%`  |
+
+#### Example
+
+```evy
+printf "The tank is %%100 full.\n\n"
+
+weather := "rainy"
+printf "It is %v today.\n" weather
+rainfall := 10
+printf "There will be %vmm of rainfall.\n" rainfall
+unicorns := false
+printf "There will be unicorns eating lollipops: %v.\n\n" unicorns
+
+printf "They said: %q\n" "Wow!"
+printf "Array: %v\n" [1 2 3]
+printf "Map: %v\n" {a:1 b:2}
+```
+
+Output
+```evy:output
+The tank is %100 full.
+
+It is rainy today.
+There will be 10mm of rainfall.
+There will be unicorns eating lollipops: false.
+
+They said: "Wow!"
+Array: [1 2 3]
+Map: {a:1 b:2}
+```
+
+#### Reference 
+
+    printf format:string a:any...
+
+The `printf` function prints its arguments to the output area according
+to the _format_ string that is the first argument. The _specifiers_
+that start with `%` and are contained in the format string are replaced
+by the remaining arguments in the given order. For example, the
+following code `printf "first: %s, second: %s" "A" "B"` prints `first:
+A, second: B`.
+
+Full list of valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| %v        | the argument in a default format  |
+| %%        | a literal percent sign `%`; consumes no value  |
```

Comment:
I would put this one first or last as it is different to the others.

Reply:
fair, i put it at the end like above. seems also more suited there to me.